### ISSUE TITLE
[WebToolsE2E][Aspire] Run a Blazor Web App ~ WebAssembly project with Aspire checked, click Restart in the dashboard, the restart is successful but a new blank page is opened.

### ISSUE THREAD

**Issue by v-reinawang** [dotnet/aspire#5978]
**INSTALL STEPS**

1. Clean machine: Win11 x64 23h2 ENU
2. Run preinstall
3. Install Dev17.12 Preview 3 [35326.43.main]
4. Install Aspire 9.0.0-preview.4.24476.9

**REPRO STEPS**
1. File > New project > Blazor Web App > Interactive render mode: WebAssembly > Interactivity location: Per page/component > Check 'Enlist in .NET Aspire orchestration' > create
2. Build and run
3. On the dashboard, click 'Restart' option behind the blazorapp resource

**Note**: This is a new feature for Aspire 9.0 P4

**ACTUAL**
The restart was successful, but a blank new page opened.
![image](https://github.com/user-attachments/assets/ce4d7aa0-010e-4071-a4c7-93a6c69a30f2)

**EXPECTED**
The restart was successful and a normal new page was opened.

**Comment by DamianEdwards** [comment]
I think this is the known VS behavior for debugging Blazor WebAssembly apps right?

**Comment by v-elenafeng** [comment]
Yes, this is the exact behavior for debugging Blazor WebAssembly apps, but a new blank page launches every time I stop/start the project.

**Comment by thaystg** [comment]
After some investigation I understood that this bug has the same root cause of this other bug: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2301374
When we click on stop or on restart on aspire, it's not terminating the debug session, with that the browser that has the Blazor app opened is not closed and the debugger tries to launch a new one to start another debugger session. In the old debugger it causes the "about:blank" page and in the new debugger it causes this popup error "unable to launch browser".
The solution for it was discussed in a group chat and it's probably terminate the old debug session before start the new one.

**Comment by LittleLittleCloud** [comment]
I can also reproduce this issue inside vscode by having an aspire apphost with two blazor wasm projects side together

You will first get a warning prompt from vscode

![Image](https://github.com/user-attachments/assets/367d3fc1-9492-41fa-b54e-98289290b980)

And after clicking `debug anyway`, a browser with one tab running a blazor page and another tab with an empty page will be opened

![Image](https://github.com/user-attachments/assets/0a9b7fde-307f-4b58-a9f3-e75ea0c4c172)

**Comment by LittleLittleCloud** [comment]
@v-reinawang This issue should be addressed in  C#DK June release, could you verify it again

**Comment by v-reinawang** [comment]
On VS Code (CDK 1.30.32) + Latest aspire 9.4.0 P1:
This issue is not repro.
![Image](https://github.com/user-attachments/assets/73238851-da50-45d1-b7e7-8eec8e8c8c21)

On VS 18.0.0 Preview 1.0 [10807.198.main] + Latest aspire 9.4.0 P1:
This issue is not repro. 
![Image](https://github.com/user-attachments/assets/04ae9068-ded9-4a6c-b1df-555d48d5a3da)

**Comment by v-reinawang** [comment]
Verify that this issue has been fixed, so close it.

**Comment by v-reinawang** [comment]
@balachir @LittleLittleCloud This issue does not repro on VS 18.0 but still repro on VS 17.14. Do I need to open a new bug for 17.14?

**Comment by LittleLittleCloud** [comment]
@v-reinawang Yes please, we probably need to port @thaystg 's most recent fix to del17.14 as well.

After talking to @thaystg , we might not backport this fix to d17.14 and the fix would be Dev18 only.

cc @JakeRadMSFT
